scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java

```diff
@@ -246,8 +263,129 @@ public SchedulerDBManager(Configuration configuration, boolean drop) {
             }
 
         });
+        
+        return new Page(lJobs, totalNbJobs);
+    }
+    
+
+    public Page<TaskState> getTaskStates(final long from, final long to, final String tag,
+            final int offset, final int limit,
+            final String user, final boolean pending, final boolean running,
+            final boolean finished, final List<SortParameter<TaskSortParameter>> sortParameters) {
+
+        DBTaskDataParameters parameters = new DBTaskDataParameters(tag, from, to, offset, limit, user,
+            pending, running, finished, sortParameters);
+        int totalNbTasks = getTotalNumberOfTasks(parameters);
+        List<TaskState> lTasks = runWithoutTransaction(TaskDBUtils.taskStateSessionWork(parameters));
+        
+        return new Page<TaskState>(lTasks, totalNbTasks);
+    }
+    
+    public Page<TaskInfo> getTasks(final long from, final long to, final String tag,
+            final int offset, final int limit,
+            final String user, final boolean pending, final boolean running,
+            final boolean finished, final List<SortParameter<TaskSortParameter>> sortParameters) {
+
+        DBTaskDataParameters parameters = new DBTaskDataParameters(tag, from, to, offset, limit, user,
+                pending, running, finished, sortParameters);
+        int totalNbTasks = getTotalNumberOfTasks(parameters);
+        List<TaskInfo> lTaskInfo =  runWithoutTransaction(TaskDBUtils.taskInfoSessionWork(parameters));
+        
+        return new Page<TaskInfo>(lTaskInfo, totalNbTasks);
+    }
+    
+    private int getTotalNumberOfTasks(final DBTaskDataParameters params) {
+
+        return runWithoutTransaction(new SessionWork<Integer>() {
+
+            @Override
+            public Integer executeWork(Session session) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "executeWork" is 14 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)